{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "0d3a313f-d827-4c45-8e5f-ea66c926ec1a",
   "metadata": {},
   "source": [
    "# Run genomic scans of Fst, pi, Dxy\n",
    "Ave/Pla samples "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "79016991-a92c-45ac-9892-1ecd8d1c2dd9",
   "metadata": {},
   "outputs": [],
   "source": [
    "module load bcftools vcftools\n",
    "export PATH=$PATH:$HOME/genomics_general:$HOME/genomics_general/VCF_processing\n",
    "baseDIR=~/snap_hap_repHZ/genome_scans\n",
    "cd $baseDIR"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6344a2be-8f0d-4884-8b1f-1776ed402f03",
   "metadata": {},
   "source": [
    "## Window estimates with genomics_general"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cea61803-92f6-465d-a5b1-726506c981cb",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Convert all Stitch-Vcfs to geno format\n",
    "baseDIR=~/snap_hap_repHZ/genome_scans\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    vcf=~/snap_hap_repHZ/Stitch/$chrom/$chrom.AvePla.stitch.SnpOnly.final.vcf.gz\n",
    "    geno=$baseDIR/geno/$chrom.AvePla.stitch.SnpOnly.final.geno\n",
    "\n",
    "    echo -e $chrom'\\n'`basename $vcf`'\\n'`basename $geno`'\\n----\\n'\n",
    "    time parseVCF.py -i $vcf -o $geno --ploidy 2 --addRefTrack\n",
    "    echo -e 'Done\\n'\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "28e9d171-a90b-4ffb-988b-0b43ee305ff4",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Scans for AvePla 10kb wins 10kb steps\n",
    "baseDIR=~/snap_hap_repHZ/genome_scans\n",
    "cd $baseDIR\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    ## chromRegion Data\n",
    "    echo -e '\\n'$chrom'\\n\\n'\n",
    "    # chrom=Chr1\n",
    "    windType=coordinate\n",
    "    windSize=10000\n",
    "    stepSize=10000\n",
    "\n",
    "    ## Population Data\n",
    "    pops=AvePla\n",
    "    popsFile=$baseDIR/samples/AvePla_pools.txt\n",
    "    pop1=Ave\n",
    "    pop2=Pla\n",
    "    \n",
    "    ## Input/Output\n",
    "    geno=$baseDIR/geno/$chrom.AvePla.stitch.SnpOnly.final.geno\n",
    "    output=$baseDIR/$pops/${pop1}-${pop2}_${chrom}_byPos_w${windSize}_s${stepSize}.csv.gz\n",
    "    threads=10\n",
    "    genoFormat=phased\n",
    "    if [ ! -d $baseDIR/$pops ]; then mkdir $baseDIR/$pops; fi\n",
    "    \n",
    "    ## Run genomic scan\n",
    "    time popgenWindows.py   -g $geno -o $output -f $genoFormat \\\n",
    "                            --windType $windType -w $windSize -s $stepSize \\\n",
    "                            -p $pop1 -p $pop2 --popsFile $popsFile --ploidy 2 \\\n",
    "                            --writeFailedWindow --addWindowID -T $threads\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fcd83b3d-a8b5-48d4-8e09-a2d6afb2ba79",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Scans for AvePla-FrYe 10kb wins 10kb steps\n",
    "baseDIR=~/snap_hap_repHZ/genome_scans\n",
    "cd $baseDIR\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    ## chromRegion Data\n",
    "    echo -e '\\n'$chrom'\\n\\n'\n",
    "    # chrom=Chr1\n",
    "    windType=coordinate\n",
    "    windSize=10000\n",
    "    stepSize=10000\n",
    "\n",
    "    ## Population Data\n",
    "    pops=AvePla-FRYe\n",
    "    popsFile=$baseDIR/samples/AvePla-FRYe_pools.txt\n",
    "    pop1=AveFR\n",
    "    pop2=AveY\n",
    "    pop3=PlaFR\n",
    "    pop4=PlaY\n",
    "\n",
    "    ## Input/Output\n",
    "    geno=$baseDIR/geno/$chrom.AvePla.stitch.SnpOnly.final.geno\n",
    "    output=$baseDIR/$pops/${pop1}-${pop2}-${pop3}-${pop4}_${chrom}_byPos_w${windSize}_s${stepSize}.csv.gz\n",
    "    threads=10\n",
    "    genoFormat=phased\n",
    "    if [ ! -d $baseDIR/$pops ]; then mkdir $baseDIR/$pops; fi\n",
    "\n",
    "    ## Run genomic scan\n",
    "    time popgenWindows.py   -g $geno -o $output -f $genoFormat \\\n",
    "                            --windType $windType -w $windSize -s $stepSize \\\n",
    "                            -p $pop1 -p $pop2 -p $pop3 -p $pop4 --popsFile $popsFile --ploidy 2 \\\n",
    "                            --writeFailedWindow --addWindowID -T $threads\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "86234227-9d21-4e0d-8dd5-c1c7ff168b9a",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Scans for AvePla-FrYe 3kb wins 300b steps\n",
    "baseDIR=~/snap_hap_repHZ/genome_scans\n",
    "cd $baseDIR\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    ## chromRegion Data\n",
    "    echo -e '\\n'$chrom'\\n\\n'\n",
    "    # chrom=Chr1\n",
    "    windType=coordinate\n",
    "    windSize=3000\n",
    "    stepSize=300\n",
    "\n",
    "    ## Population Data\n",
    "    pops=AvePla-FRYe\n",
    "    popsFile=$baseDIR/samples/AvePla-FRYeHybrid_pools.txt\n",
    "    pop1=AveFR\n",
    "    pop2=AveY\n",
    "    pop3=PlaFR\n",
    "    pop4=PlaY\n",
    "\n",
    "    ## Input/Output\n",
    "    geno=$baseDIR/geno/$chrom.AvePla.stitch.SnpOnly.final.geno\n",
    "    output=$baseDIR/$pops/${pop1}-${pop2}-${pop3}-${pop4}_${chrom}_byPos_w${windSize}_s${stepSize}.csv.gz\n",
    "    threads=8\n",
    "    genoFormat=phased\n",
    "    if [ ! -d $baseDIR/$pops ]; then mkdir $baseDIR/$pops; fi\n",
    "\n",
    "    ## Run genomic scan\n",
    "    time popgenWindows.py   -g $geno -o $output -f $genoFormat \\\n",
    "                            --windType $windType -w $windSize -s $stepSize \\\n",
    "                            -p $pop1 -p $pop2 -p $pop3 -p $pop4 --popsFile $popsFile --ploidy 2 \\\n",
    "                            --writeFailedWindow --addWindowID -T $threads\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d347e71a-4544-4691-9a5a-1065b0f9cda2",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Scans for AvePla-FrYe 3kb wins 3kb steps\n",
    "baseDIR=~/snap_hap_repHZ/genome_scans\n",
    "cd $baseDIR\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    ## chromRegion Data\n",
    "    echo -e '\\n'$chrom'\\n\\n'\n",
    "    # chrom=Chr1\n",
    "    windType=coordinate\n",
    "    windSize=3000\n",
    "    stepSize=3000\n",
    "\n",
    "    ## Population Data\n",
    "    pops=AvePla-FRYe\n",
    "    popsFile=$baseDIR/samples/AvePla-FRYeHybrid_pools.txt\n",
    "    pop1=AveFR\n",
    "    pop2=AveY\n",
    "    pop3=PlaFR\n",
    "    pop4=PlaY\n",
    "\n",
    "    ## Input/Output\n",
    "    geno=$baseDIR/geno/$chrom.AvePla.stitch.SnpOnly.final.geno\n",
    "    output=$baseDIR/$pops/${pop1}-${pop2}-${pop3}-${pop4}_${chrom}_byPos_w${windSize}_s${stepSize}.csv.gz\n",
    "    threads=8\n",
    "    genoFormat=phased\n",
    "    if [ ! -d $baseDIR/$pops ]; then mkdir $baseDIR/$pops; fi\n",
    "\n",
    "    ## Run genomic scan\n",
    "    time popgenWindows.py   -g $geno -o $output -f $genoFormat \\\n",
    "                            --windType $windType -w $windSize -s $stepSize \\\n",
    "                            -p $pop1 -p $pop2 -p $pop3 -p $pop4 --popsFile $popsFile --ploidy 2 \\\n",
    "                            --writeFailedWindow --addWindowID -T $threads\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8fc08614-46a6-4c2d-a5db-9f641cda0fd8",
   "metadata": {},
   "source": [
    "## Fst per site (vcftools)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d2faf745-71df-495d-9a23-3572b3210c16",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fst\n",
    "baseDIR=~/snap_hap_repHZ/genome_scans\n",
    "\n",
    "## Populations\n",
    "pop_Ave=~/snap_hap_repHZ/genome_scans/samples/samples_Ave.txt\n",
    "pop_AveY=~/snap_hap_repHZ/genome_scans/samples/samples_AveY.txt\n",
    "pop_AveFR=~/snap_hap_repHZ/genome_scans/samples/samples_AveFR.txt\n",
    "pop_Pla=~/snap_hap_repHZ/genome_scans/samples/samples_Pla.txt\n",
    "pop_PlaY=~/snap_hap_repHZ/genome_scans/samples/samples_PlaY.txt\n",
    "pop_PlaFR=~/snap_hap_repHZ/genome_scans/samples/samples_PlaFR.txt\n",
    "\n",
    "## Vcf\n",
    "inVcf=~/snap_hap_repHZ/Stitch/AvePla.all.stitch.SnpOnly.final.sorted.vcf.gz\n",
    "\n",
    "## Run VCFTOOLS\n",
    "\n",
    "# Ave vs Pla\n",
    "output=$baseDIR/AvePla/Ave-Pla.site\n",
    "vcftools    --gzvcf $inVcf \\\n",
    "            --weir-fst-pop $pop_Ave \\\n",
    "            --weir-fst-pop $pop_Pla \\\n",
    "            --out $output\n",
    "\n",
    "# AveFR vs AveY\n",
    "output=$baseDIR/AvePla-FRYe/AveFR-AveY.site\n",
    "vcftools    --gzvcf $inVcf \\\n",
    "            --weir-fst-pop $pop_AveFR \\\n",
    "            --weir-fst-pop $pop_AveY \\\n",
    "            --out $output\n",
    "            \n",
    "# AveFR vs PlaFR\n",
    "output=$baseDIR/AvePla-FRYe/AveFR-PlaFR.site\n",
    "vcftools    --gzvcf $inVcf \\\n",
    "            --weir-fst-pop $pop_AveFR \\\n",
    "            --weir-fst-pop $pop_PlaFR \\\n",
    "            --out $output\n",
    "\n",
    "# AveFR vs PlaY\n",
    "output=$baseDIR/AvePla-FRYe/AveFR-PlaY.site\n",
    "vcftools    --gzvcf $inVcf \\\n",
    "            --weir-fst-pop $pop_AveFR \\\n",
    "            --weir-fst-pop $pop_PlaY \\\n",
    "            --out $output\n",
    "\n",
    "## AveY vs PlaFR\n",
    "output=$baseDIR/AvePla-FRYe/AveY-PlaFR.site\n",
    "vcftools    --gzvcf $inVcf \\\n",
    "            --weir-fst-pop $pop_AveY \\\n",
    "            --weir-fst-pop $pop_PlaFR \\\n",
    "            --out $output\n",
    "\n",
    "## AveY vs PlaY\n",
    "output=$baseDIR/AvePla-FRYe/AveY-PlaY.site\n",
    "vcftools    --gzvcf $inVcf \\\n",
    "            --weir-fst-pop $pop_AveY \\\n",
    "            --weir-fst-pop $pop_PlaY \\\n",
    "            --out $output\n",
    "\n",
    "## PlaFR vs PlaY\n",
    "output=$baseDIR/AvePla-FRYe/PlaFR-PlaY.site\n",
    "vcftools    --gzvcf $inVcf \\\n",
    "            --weir-fst-pop $pop_PlaFR \\\n",
    "            --weir-fst-pop $pop_PlaY \\\n",
    "            --out $output"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5c97b683-547b-4ed6-84a1-9edfd1dc68ea",
   "metadata": {},
   "source": [
    "## Allele Frequency (vcftools)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dbe6bda1-dd90-48bc-a17b-7e6ecf82313c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Avellanet\n",
    "cd ~/snap_hap_repHZ/genome_scans/alleleFreqs\n",
    "\n",
    "## -----\n",
    "## Ave\n",
    "pop=Ave\n",
    "sampleList=~/snap_hap_repHZ/genome_scans/samples/samples_$pop.txt\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    echo -e $pop'\\t'$chrom'\\n'\n",
    "    inVCF=~/snap_hap_repHZ/statphase/AvePla_FrYe/$chrom.AvePla.FrYe.sorted.statphased.vcf.gz\n",
    "    output=~/snap_hap_repHZ/genome_scans/alleleFreqs/$chrom.$pop\n",
    "    time vcftools --gzvcf $inVCF --keep $sampleList --freq2 --out $output\n",
    "    gzip $output.frq\n",
    "done\n",
    "\n",
    "## -----\n",
    "## AveFR\n",
    "pop=AveFR\n",
    "sampleList=~/snap_hap_repHZ/genome_scans/samples/samples_$pop.txt\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    echo -e $pop'\\t'$chrom'\\n'\n",
    "    inVCF=~/snap_hap_repHZ/statphase/AvePla_FrYe/$chrom.AvePla.FrYe.sorted.statphased.vcf.gz\n",
    "    output=~/snap_hap_repHZ/genome_scans/alleleFreqs/$chrom.$pop\n",
    "    time vcftools --gzvcf $inVCF --keep $sampleList --freq2 --out $output\n",
    "    gzip $output.frq\n",
    "done\n",
    "\n",
    "## -----\n",
    "## AveY\n",
    "pop=AveY\n",
    "sampleList=~/snap_hap_repHZ/genome_scans/samples/samples_$pop.txt\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    echo -e $pop'\\t'$chrom'\\n'\n",
    "    inVCF=~/snap_hap_repHZ/statphase/AvePla_FrYe/$chrom.AvePla.FrYe.sorted.statphased.vcf.gz\n",
    "    output=~/snap_hap_repHZ/genome_scans/alleleFreqs/$chrom.$pop\n",
    "    time vcftools --gzvcf $inVCF --keep $sampleList --freq2 --out $output\n",
    "    gzip $output.frq\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "afa4101d-2147-4d7d-be44-f68fdae7a664",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Planoles\n",
    "cd ~/snap_hap_repHZ/genome_scans/alleleFreqs\n",
    "\n",
    "## -----\n",
    "## Pla\n",
    "pop=Pla\n",
    "sampleList=~/snap_hap_repHZ/genome_scans/samples/samples_$pop.txt\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    echo -e $pop'\\t'$chrom'\\n'\n",
    "    inVCF=~/snap_hap_repHZ/statphase/AvePla_FrYe/$chrom.AvePla.FrYe.sorted.statphased.vcf.gz\n",
    "    output=~/snap_hap_repHZ/genome_scans/alleleFreqs/$chrom.$pop\n",
    "    time vcftools --gzvcf $inVCF --keep $sampleList --freq2 --out $output\n",
    "    gzip $output.frq\n",
    "done\n",
    "\n",
    "## -----\n",
    "## PlaFR\n",
    "pop=PlaFR\n",
    "sampleList=~/snap_hap_repHZ/genome_scans/samples/samples_$pop.txt\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    echo -e $pop'\\t'$chrom'\\n'\n",
    "    inVCF=~/snap_hap_repHZ/statphase/AvePla_FrYe/$chrom.AvePla.FrYe.sorted.statphased.vcf.gz\n",
    "    output=~/snap_hap_repHZ/genome_scans/alleleFreqs/$chrom.$pop\n",
    "    time vcftools --gzvcf $inVCF --keep $sampleList --freq2 --out $output\n",
    "    gzip $output.frq\n",
    "done\n",
    "\n",
    "## -----\n",
    "## PlaY\n",
    "pop=PlaY\n",
    "sampleList=~/snap_hap_repHZ/genome_scans/samples/samples_$pop.txt\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    echo -e $pop'\\t'$chrom'\\n'\n",
    "    inVCF=~/snap_hap_repHZ/statphase/AvePla_FrYe/$chrom.AvePla.FrYe.sorted.statphased.vcf.gz\n",
    "    output=~/snap_hap_repHZ/genome_scans/alleleFreqs/$chrom.$pop\n",
    "    time vcftools --gzvcf $inVCF --keep $sampleList --freq2 --out $output\n",
    "    gzip $output.frq\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e84792a1-9d9a-4870-adef-97ed294b18ad",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true
   },
   "source": [
    "## HWE or heterozygosity per site _(vcftools)_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "50af5504-f6f3-4c4b-a848-116acb952406",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Avellanet\n",
    "cd ~/snap_hap_repHZ/genome_scans/hardy\n",
    "\n",
    "## -----\n",
    "## Ave\n",
    "pop=Ave\n",
    "sampleList=~/snap_hap_repHZ/genome_scans/samples/samples_$pop.txt\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    echo -e '\\n-----\\n'$pop'\\t'$chrom'\\n-----\\n'\n",
    "    inVCF=~/snap_hap_repHZ/statphase/AvePla_FrYe/$chrom.AvePla.FrYe.sorted.statphased.vcf.gz\n",
    "    output=~/snap_hap_repHZ/genome_scans/hardy/$chrom.$pop\n",
    "    time vcftools --gzvcf $inVCF --keep $sampleList --hardy --out $output\n",
    "    gzip $output.hwe\n",
    "done\n",
    "\n",
    "## -----\n",
    "## AveM\n",
    "pop=AveM\n",
    "sampleList=~/snap_hap_repHZ/genome_scans/samples/samples_$pop.txt\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    echo -e '\\n-----\\n'$pop'\\t'$chrom'\\n-----\\n'\n",
    "    inVCF=~/snap_hap_repHZ/statphase/AvePla_FrYe/$chrom.AvePla.FrYe.sorted.statphased.vcf.gz\n",
    "    output=~/snap_hap_repHZ/genome_scans/hardy/$chrom.$pop\n",
    "    time vcftools --gzvcf $inVCF --keep $sampleList --hardy --out $output\n",
    "    gzip $output.hwe\n",
    "done\n",
    "\n",
    "## -----\n",
    "## AveY\n",
    "pop=AveY\n",
    "sampleList=~/snap_hap_repHZ/genome_scans/samples/samples_$pop.txt\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    echo -e '\\n-----\\n'$pop'\\t'$chrom'\\n-----\\n'\n",
    "    inVCF=~/snap_hap_repHZ/statphase/AvePla_FrYe/$chrom.AvePla.FrYe.sorted.statphased.vcf.gz\n",
    "    output=~/snap_hap_repHZ/genome_scans/hardy/$chrom.$pop\n",
    "    time vcftools --gzvcf $inVCF --keep $sampleList --hardy --out $output\n",
    "    gzip $output.hwe\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3cdbdaba-9c3e-49ee-8a0c-8c23806f9a14",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Planoles\n",
    "cd ~/snap_hap_repHZ/genome_scans/hardy\n",
    "\n",
    "## -----\n",
    "## Pla\n",
    "pop=Pla\n",
    "sampleList=~/snap_hap_repHZ/genome_scans/samples/samples_$pop.txt\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    echo -e '\\n-----\\n'$pop'\\t'$chrom'\\n-----\\n'\n",
    "    inVCF=~/snap_hap_repHZ/statphase/AvePla_FrYe/$chrom.AvePla.FrYe.sorted.statphased.vcf.gz\n",
    "    output=~/snap_hap_repHZ/genome_scans/hardy/$chrom.$pop\n",
    "    time vcftools --gzvcf $inVCF --keep $sampleList --hardy --out $output\n",
    "    gzip $output.hwe\n",
    "done\n",
    "\n",
    "## -----\n",
    "## PlaM\n",
    "pop=PlaM\n",
    "sampleList=~/snap_hap_repHZ/genome_scans/samples/samples_$pop.txt\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    echo -e '\\n-----\\n'$pop'\\t'$chrom'\\n-----\\n'\n",
    "    inVCF=~/snap_hap_repHZ/statphase/AvePla_FrYe/$chrom.AvePla.FrYe.sorted.statphased.vcf.gz\n",
    "    output=~/snap_hap_repHZ/genome_scans/hardy/$chrom.$pop\n",
    "    time vcftools --gzvcf $inVCF --keep $sampleList --hardy --out $output\n",
    "    gzip $output.hwe\n",
    "done\n",
    "\n",
    "## -----\n",
    "## PlaY\n",
    "pop=PlaY\n",
    "sampleList=~/snap_hap_repHZ/genome_scans/samples/samples_$pop.txt\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    echo -e '\\n-----\\n'$pop'\\t'$chrom'\\n-----\\n'\n",
    "    inVCF=~/snap_hap_repHZ/statphase/AvePla_FrYe/$chrom.AvePla.FrYe.sorted.statphased.vcf.gz\n",
    "    output=~/snap_hap_repHZ/genome_scans/hardy/$chrom.$pop\n",
    "    time vcftools --gzvcf $inVCF --keep $sampleList --hardy --out $output\n",
    "    gzip $output.hwe\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6e61c34b-bd78-4df4-a302-d622d706beb5",
   "metadata": {},
   "source": [
    "## Site Pi _(vcftools)_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "87a5dd4f-36e1-4e1d-acef-0bfcb076f161",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Avellanet\n",
    "cd ~/snap_hap_repHZ/genome_scans/sitePi\n",
    "\n",
    "## -----\n",
    "## Ave\n",
    "pop=Ave\n",
    "sampleList=~/snap_hap_repHZ/genome_scans/samples/samples_$pop.txt\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    echo -e '\\n-----\\n'$pop'\\t'$chrom'\\n-----\\n'\n",
    "    inVCF=~/snap_hap_repHZ/statphase/AvePla_FrYe/$chrom.AvePla.FrYe.sorted.statphased.vcf.gz\n",
    "    output=~/snap_hap_repHZ/genome_scans/sitePi/$chrom.$pop\n",
    "    time vcftools --gzvcf $inVCF --keep $sampleList --site-pi --out $output\n",
    "    gzip $output.sites.pi\n",
    "done\n",
    "\n",
    "## -----\n",
    "## AveM\n",
    "pop=AveM\n",
    "sampleList=~/snap_hap_repHZ/genome_scans/samples/samples_$pop.txt\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    echo -e '\\n-----\\n'$pop'\\t'$chrom'\\n-----\\n'\n",
    "    inVCF=~/snap_hap_repHZ/statphase/AvePla_FrYe/$chrom.AvePla.FrYe.sorted.statphased.vcf.gz\n",
    "    output=~/snap_hap_repHZ/genome_scans/sitePi/$chrom.$pop\n",
    "    time vcftools --gzvcf $inVCF --keep $sampleList --site-pi --out $output\n",
    "    gzip $output.sites.pi\n",
    "done\n",
    "\n",
    "## -----\n",
    "## AveY\n",
    "pop=AveY\n",
    "sampleList=~/snap_hap_repHZ/genome_scans/samples/samples_$pop.txt\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    echo -e '\\n-----\\n'$pop'\\t'$chrom'\\n-----\\n'\n",
    "    inVCF=~/snap_hap_repHZ/statphase/AvePla_FrYe/$chrom.AvePla.FrYe.sorted.statphased.vcf.gz\n",
    "    output=~/snap_hap_repHZ/genome_scans/sitePi/$chrom.$pop\n",
    "    time vcftools --gzvcf $inVCF --keep $sampleList --site-pi --out $output\n",
    "    gzip $output.sites.pi\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3527a8bc-4b8b-4925-b31e-e1e471085917",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Planoles\n",
    "cd ~/snap_hap_repHZ/genome_scans/sitePi\n",
    "\n",
    "## -----\n",
    "## Pla\n",
    "pop=Pla\n",
    "sampleList=~/snap_hap_repHZ/genome_scans/samples/samples_$pop.txt\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    echo -e '\\n-----\\n'$pop'\\t'$chrom'\\n-----\\n'\n",
    "    inVCF=~/snap_hap_repHZ/statphase/AvePla_FrYe/$chrom.AvePla.FrYe.sorted.statphased.vcf.gz\n",
    "    output=~/snap_hap_repHZ/genome_scans/sitePi/$chrom.$pop\n",
    "    time vcftools --gzvcf $inVCF --keep $sampleList --site-pi --out $output\n",
    "    gzip $output.sites.pi\n",
    "done\n",
    "\n",
    "## -----\n",
    "## PlaM\n",
    "pop=PlaM\n",
    "sampleList=~/snap_hap_repHZ/genome_scans/samples/samples_$pop.txt\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    echo -e '\\n-----\\n'$pop'\\t'$chrom'\\n-----\\n'\n",
    "    inVCF=~/snap_hap_repHZ/statphase/AvePla_FrYe/$chrom.AvePla.FrYe.sorted.statphased.vcf.gz\n",
    "    output=~/snap_hap_repHZ/genome_scans/sitePi/$chrom.$pop\n",
    "    time vcftools --gzvcf $inVCF --keep $sampleList --site-pi --out $output\n",
    "    gzip $output.sites.pi\n",
    "done\n",
    "\n",
    "## -----\n",
    "## PlaY\n",
    "pop=PlaY\n",
    "sampleList=~/snap_hap_repHZ/genome_scans/samples/samples_$pop.txt\n",
    "for chrom in Chr{1..8}\n",
    "do\n",
    "    echo -e '\\n-----\\n'$pop'\\t'$chrom'\\n-----\\n'\n",
    "    inVCF=~/snap_hap_repHZ/statphase/AvePla_FrYe/$chrom.AvePla.FrYe.sorted.statphased.vcf.gz\n",
    "    output=~/snap_hap_repHZ/genome_scans/sitePi/$chrom.$pop\n",
    "    time vcftools --gzvcf $inVCF --keep $sampleList --site-pi --out $output\n",
    "    gzip $output.sites.pi\n",
    "done"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
